tests/Tests.FeatureManagement/FeatureManagement.cs

```diff
@@ -242,6 +242,115 @@ public async Task Percentage()
             Assert.True(enabledCount > 0 && enabledCount < 10);
         }
 
+        [Fact]
+        public async Task Targeting()
+        {
+            IConfiguration config = new ConfigurationBuilder().AddJsonFile("appsettings.json").Build();
+
+            var services = new ServiceCollection();
+
+            services
+                .Configure<FeatureManagementOptions>(options =>
+                {
+                    options.IgnoreMissingFeatureFilters = true;
+                });
+
+            services
+                .AddSingleton(config)
+                .AddFeatureManagement()
+                .AddFeatureFilter<ContextualTargetingFilter>();
+
+            ServiceProvider serviceProvider = services.BuildServiceProvider();
+
+            IFeatureManager featureManager = serviceProvider.GetRequiredService<IFeatureManager>();
+
+            string beta = Enum.GetName(typeof(Features), Features.Beta);
+
+            //
+            // Targeted by user id
+            Assert.True(await featureManager.IsEnabledAsync(beta, new TargetingContext
+            {
+                UserId = "Jeff"
+            }));
+
+            //
+            // Not targeted by user id, but targeted by default rollout
+            Assert.True(await featureManager.IsEnabledAsync(beta, new TargetingContext
+            {
+                UserId = "Anne"
+            }));
```

Comment:
Wow, if we change the hash algorithm in the future, this one will be broken :)

Reply:
Yep, but I think it should break. It would be a breaking change to end users after all, and that is something we want to catch with tests.